pymeasure/adapters/telnet.py

```diff
@@ -0,0 +1,80 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2020 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import telnetlib
+import time
+
+from .adapter import Adapter
+
+
+class TelnetAdapter(Adapter):
+    """ Adapter class for using the Python telnetlib package to allow
+    communication to instruments
+
+    :param host: host address of the instrument
+    :param port: TCPIP port
+    :param query_delay: delay in seconds between write and read in the ask
+                        method
+    :param kwargs: Any valid key-word argument for telnetlib.Telnet
+    """
+
+    def __init__(self, host, port=0, query_delay=0, **kwargs):
+        self.query_delay = query_delay
+        self.connection = telnetlib.Telnet(host, port, **kwargs)
```

Comment:
I can do that, but do consider this more a limitation than a feature. I guess this is not the right place to criticise VISAAdapter, but many more keyword arguments than in your safelist are possible depending on the connection type of VISA. One such limitation I encountered recently was when using a serial device where one can not specify the `baud_rate` as keyword argument. But adding `baud_rate` to the safe keyword list would be likely problematic since not every connection type will allow such a keyword... So in the end only `get_instrument` knows which keywords are actually allowed

For the case of [`Telnet`](https://docs.python.org/3.9/library/telnetlib.html#telnetlib.Telnet) in fact only one additional parameter (`timeout`) seems allowed.